fendermint/vm/topdown/src/voting/mod.rs

```diff
@@ -148,8 +156,8 @@ where
     pub fn add_block(
         &self,
         block_height: BlockHeight,
-        block_hash: Option<V>,
-    ) -> StmResult<(), Error<K>> {
+        payload: Option<TopdownVote>,
```

Comment:
I dont think `add_block` is ever needed, vote tally should directly get the data from `syncer` cache instead of storing its own data. Since `syncer` cache will be persisted, it will be much more deterministic then something in memory. Mark it as `todo` maybe.